src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -453,6 +453,63 @@ public function isAbsolutePath($file)
     }
 
     /**
+     * Creates a temporary file with support for custom stream wrappers.
+     *
+     * @param string $dir    The directory where the temporary filename will be created.
+     * @param string $prefix The prefix of the generated temporary filename.
+     *                       Note: Windows uses only the first three characters of prefix.
+     *
+     * @return string The new temporary filename (with path), or false on failure.
+     */
+    public function tempNam($dir, $prefix)
+    {
+        $limit = 10;
+        $schemeSeparator = '://';
+        $scheme = $this->getScheme($dir);
+        $hierarchy = $this->getHierarchy($dir);
+
+        // If no scheme or scheme is "file" create temp file in local filesystem
+        if (false === $scheme || 'file' === $scheme) {
+            $tmpFile = tempnam($hierarchy, $prefix);
+        }
+
+        // If no scheme we're done, just return the file generated by tempnam
+        if (false === $scheme) {
+            return $tmpFile;
+        }
+
+        // If scheme is "file" add this back onto whatever tempnam created
+        if ('file' === $scheme) {
+            // Return the version with scheme
+            return $scheme.$schemeSeparator.$tmpFile;
+        }
+
+        // Loop until we create a valid temp file or have reached $limit attempts
+        for ($i = 0; $i < $limit; $i++) {
+
+            // Create a unique filename
+            $tmpFile = $dir.'/'.$prefix.uniqid(mt_rand(), true);
+
+            // If the file already exists restart the loop
+            // Use fopen instead of file_exists as some streams don't support stat
+            if (false !== @fopen($tmpFile, 'r')) {
```

Comment:
you need to close the stream after opening it, otherwise you will be leaking stream resources